planning/behavior_path_planner/src/scene_module/goal_planner/goal_planner_module.cpp

```diff
@@ -888,22 +879,50 @@ void GoalPlannerModule::decideVelocity()
   }
 }
 
-CandidateOutput GoalPlannerModule::planCandidate() const
+BehaviorModuleOutput GoalPlannerModule::planPullOver()
 {
-  return CandidateOutput(
-    thread_safe_data_.get_pull_over_path() ? thread_safe_data_.get_pull_over_path()->getFullPath()
-                                           : PathWithLaneId());
+  if (!hasDecidedPath()) {
+    return planPullOverAsCandidate();
+  }
+
+  return planPullOverAsOutput();
 }
 
-BehaviorModuleOutput GoalPlannerModule::planWithGoalModification()
+BehaviorModuleOutput GoalPlannerModule::planPullOverAsCandidate()
 {
   // if pull over path candidates generation is not finished, use previous module output
   if (thread_safe_data_.get_pull_over_path_candidates().empty()) {
     return getPreviousModuleOutput();
   }
 
-  resetPathCandidate();
-  resetPathReference();
+  BehaviorModuleOutput output{};
+  const BehaviorModuleOutput pull_over_output = planPullOverAsOutput();
+  output.modified_goal = pull_over_output.modified_goal;
+  output.path = std::make_shared<PathWithLaneId>(generateStopPath());
+
+  const auto target_drivable_lanes = utils::getNonOverlappingExpandedLanes(
+    *output.path, generateDrivableLanes(), planner_data_->drivable_area_expansion_parameters);
+
+  DrivableAreaInfo current_drivable_area_info{};
+  current_drivable_area_info.drivable_lanes = target_drivable_lanes;
+  output.drivable_area_info = utils::combineDrivableAreaInfo(
+    current_drivable_area_info, getPreviousModuleOutput().drivable_area_info);
+
+  if (!thread_safe_data_.foundPullOverPath()) {
+    return output;
+  }
+
+  path_candidate_ = pull_over_output.path;
+
+  return output;
+}
+
+BehaviorModuleOutput GoalPlannerModule::planPullOverAsOutput()
+{
+  // if pull over path candidates generation is not finished, use previous module output
+  if (thread_safe_data_.get_pull_over_path_candidates().empty()) {
+    return getPreviousModuleOutput();
+  }
 
```

Comment:
this check can be moved in the beginning of `planPullOver()` function or somewhere?
this is implemented in the same way in `planPullOverAsCandidate()`

Reply:
this is needed here for the case from `planPullOverAsCandidate` is called `planWaitingApproval`